gui/default/syncthing/folder/browseController.js

```diff
@@ -0,0 +1,56 @@
+angular.module('syncthing.folder')
+    .controller('BrowseController', function (
+        $scope,
+        CurrentFolder,
+        Ignores,
+        Browse,
+        FileMatches,
+    ) {
+        'use strict';
+
+        // Bind methods directly to the controller so we can use controllerAs in template
+        var self = this;
+
+        // public definitions
+
+        self.folder = CurrentFolder;
+        // Reference to browse data for the current folder
+        self.browse = undefined;
+        self.fileMatches = undefined;
+
+        $scope.$watch(function() {
+            return self.folder.id;
+        }, function (newId) {
+            if (newId) {
+                self.browse = Browse.forFolder(newId);
+                self.fileMatches = FileMatches.forFolder(newId);
```

Comment:
This `forFolder` pattern made sense back when I thought there would be a need to cache data for multiple folders at once but there is not. It has become repetitive and verbose, clumsily used in different ways for each service. I will remove this so the service holds just one dataset at a time. If there's ever a need to cache or separate the data it can be added in a more object oriented fashion.